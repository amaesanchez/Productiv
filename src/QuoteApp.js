import React, { useState } from "react";
import axios from "axios";
import Quote from "./Quote";

const BASE_URL = "https://inspo-quotes-api.herokuapp.com/quotes";

/** Componenent for Quote and button to request a quote
 * Uses ajax request to fetc random quote
 *
 * State
 *  - quote
 *
 * App -> QuoteApp -> Quote
 */

function QuoteApp() {
  const [quote, setQuote] = useState({ text: "", author: "" });

  async function getQuote() {
    const resp = await axios.get(`${BASE_URL}/random`);
    setQuote(resp.data.quote);
  }

  return (
    <div className="QuoteApp">
      {quote.text.length !== 0 && <Quote quote={quote} />}
      <button
        className="quoteBtn mb-3 btn btn-light text-secondary"
        onClick={getQuote}
      >
        {quote.text.length === 0
          ? "Click here for an inspirational quote!"
          : "New quote"}
      </button>
    </div>
  );
}

export default QuoteApp;
