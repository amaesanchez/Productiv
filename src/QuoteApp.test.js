import { fireEvent, render, screen } from "@testing-library/react";
import QuoteApp from "./QuoteApp";

describe("QuoteApp", function () {
  it("renders without crashing", function () {
    render(<QuoteApp />);
  });

  it("matches snapshot", function () {
    const { container } = render(<QuoteApp />);
    expect(container).toMatchSnapshot();
  });

  it("Renders button for fetching quote", function () {
    const { container } = render(<QuoteApp />);

    const quoteBtn = container.querySelector(".quoteBtn")
    expect(quoteBtn).toBeInTheDocument();
  });

  //TODO: figure out why button callback doesn't seem to be working

  // it("Renders quote after button is clicked", function () {
  //   const { container, debug } = render(<QuoteApp />);
  //   const getQuote = jest.fn()

  //   const quoteBtn = screen.getByText("Click here for an inspirational quote!")
  //   debug(container);
  //   fireEvent.click(quoteBtn)

  //   expect(getQuote).toBeCalled();
  // });
});
