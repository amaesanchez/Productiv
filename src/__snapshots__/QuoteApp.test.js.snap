// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteApp matches snapshot 1`] = `
<div>
  <div
    class="QuoteApp"
  >
    <button
      class="quoteBtn mb-3 btn btn-light text-secondary"
    >
      Click here for an inspirational quote!
    </button>
  </div>
</div>
`;
